packages/grafana-ui/src/components/LLMQueryEditor/LLMQueryEditor.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState } from 'react';
+
+import { Button } from '../Button';
+import { Field } from '../Forms/Field';
+import { Form } from '../Forms/Form';
+import { Input } from '../Input/Input';
+
+interface LLMSrvChatMessage {
+  role: string;
+  content: string;
+}
+
+interface LLMSrvChatCompletionRequest {
+  model?: string;
+  messages: LLMSrvChatMessage[];
+}
+
+// We can't import the real interface from @grafana/runtime, but
+// we can use this subset. Hooray for structural typing!
+interface LLMSrv {
+  chatCompletions(request: LLMSrvChatCompletionRequest): Promise<string>;
+}
+
+export interface LLMQueryEditorProps {
+  systemPrompt: string;
+  onChange: (text: string) => void;
+  llmSrv: LLMSrv; // Can't import @grafana/runtime here
+}
+
+export function LLMQueryEditor({ systemPrompt, onChange, llmSrv }: LLMQueryEditorProps): JSX.Element {
+  const [query, setQuery] = useState('');
+  return (
+    <div>
+      <Form
+        onSubmit={async ({ prompt }) => {
+          const returnedMessage = await llmSrv.chatCompletions({
+            messages: [
+              { role: 'system', content: systemPrompt },
+              { role: 'user', content: prompt },
+            ],
+          });
+          setQuery(returnedMessage);
+          onChange(returnedMessage);
+        }}
+      >
+        {({ register }) => {
+          return (
+            <>
+              <Field>
+                <Input {...register('prompt')} />
+              </Field>
+              <Button type="submit">Translate</Button>
+            </>
+          );
+        }}
+      </Form>
+      {query !== '' && <div>{query}</div>}
+    </div>
+  );
```

Comment:
I have a couple of ideas on how to improve this:

1. Instead of doing an async call inline you can use the [`useAsyncFn`](https://github.com/streamich/react-use/blob/master/docs/useAsyncFn.md) and it will make sure to handle all the edge cases with component being unmounted when async call returns etc. It will also make it possible to manage the loading state if needed.

```suggestion
export function LLMQueryEditor({ systemPrompt, onChange, llmSrv }: LLMQueryEditorProps): JSX.Element {
  // might not be needed if you instead pass the expr as a prop to the editor.
  const [query, setQuery] = useState('');
  const [state, onSubmit] = useAsyncFn(({ prompt }) => {
     const returnedMessage = await llmSrv.chatCompletions({
            messages: [
              { role: 'system', content: systemPrompt },
              { role: 'user', content: prompt },
            ],
          });
     setQuery(returnedMessage);
     onChange(returnedMessage);
  }, [systemPrompt, onChange]);
  
  return (
    <div>
      <Form
        onSubmit={onSubmit}
      >
        {({ register }) => {
          return (
            <>
              <Field>
                <Input {...register('prompt')} />
              </Field>
              <Button type="submit">Translate</Button>
            </>
          );
        }}
      </Form>
      {query !== '' && <div>{query}</div>}
    </div>
  );
```

Reply:
Ah nice, added. Now we have a loading spinner too! 🎉